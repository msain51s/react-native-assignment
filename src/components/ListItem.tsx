import React from "react";
import FormTextView from "./FormTextView";
import styles from "../utility/Styles";
import {View} from 'react-native'

export type ListItemProps = {
    name:string,
    occupation:string,
    company:string,
    emailAddress:string,
    phoneNumber:string,
    linkedInUrl:string
}

function ListItem({data}:{data:ListItemProps}){
    return(
        <View style={styles.cardStyle}>
          <FormTextView title='Name : ' value={data.name}/>
          <FormTextView title='Occupation : ' value={data.occupation}/>
          <FormTextView title='Company : ' value={data.company}/>
          <FormTextView title='Email Address : ' value={data.emailAddress}/>
          <FormTextView title='Phone Number : ' value={data.phoneNumber}/>
          <FormTextView title='LinkedInUrl : ' value={data.linkedInUrl}/> 
          </View>
    )
}


export default ListItem