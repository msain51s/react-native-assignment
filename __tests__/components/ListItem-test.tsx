import React from "react";
import renderer from 'react-test-renderer'
import ListItem, { ListItemProps } from "../../src/components/ListItem";

const mockProps : ListItemProps ={
    name:'abc',
    occupation:'Engineer',
    company:'xyz',
    emailAddress:'xyz.com',
    phoneNumber:'1234567890',
    linkedInUrl:'mm.com'
}
test('ListItem renders correctly',()=>{
    
    const tree = renderer.create(
    <ListItem 
    data={mockProps}
    />)

    expect(tree).toMatchSnapshot()
})