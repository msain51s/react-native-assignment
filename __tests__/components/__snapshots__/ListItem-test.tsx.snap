// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 8,
      "margin": 10,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Name : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      abc
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Occupation : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Engineer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Company : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      xyz
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Email Address : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      xyz.com
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      Phone Number : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      1234567890
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      LinkedInUrl : 
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "margin": 10,
        }
      }
    >
      mm.com
    </Text>
  </View>
</View>
`;
